### PR TITLE
Several changes were implemented in AED and APAC transform modules

### PR DESCRIPTION
**Pull request by Maik8080**
Hi Plartoo,

I did greatest changes implemented on the AED and APAC transform modules, please be carefull with these changes since they are necessary to work with the latest data.

Have a good day!

### CODE REVIEW

File: data_architect_misc/data_transformer/configs/AED_Ukraine/config.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "__comment__": "E.g., >> python transform.py -c configs/common_qa_config.json -i ./path/to/input/file/input_data_file.xlsx",
 
     "__comment__": "(Optional) Enter the sheet name of the input Excel file to read data from. Default is the first sheet.",
-    "input_sheet_name": "DateTemplate",
+    "input_sheet_name": "DataTemplate",
```
Comment (plartoo) [inline data_architect_misc/data_transformer/configs/AED_Ukraine/config.json:9]: If there is only one sheet in Ukraine datafiles, we can delete this "input_sheet_name" key in the config file and the transform module will read the sheet_index=0 by default. Maybe that's a more robust approach?

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "write_output": true,
     "output_folder_path": "./output/APAC_Taiwan/",
     "output_file_name_prefix": "Transformed_Taiwan_20200501_20200531_",
+    "output_encoding" : "utf-8-sig",
```
Comment (plartoo) [inline data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json:15]: Can we just leave it to "utf-8"? Does it work? If so, please leave it to "utf-8" because I'm afraid using "utf-8-sig" will create issues when we upload this file to SQL Server.

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "write_output": true,
     "output_folder_path": "./output/APAC_Taiwan/",
     "output_file_name_prefix": "Transformed_Taiwan_20200501_20200531_",
+    "output_encoding" : "utf-8-sig",
```
Comment (plartoo) [inline data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json:15]: Same as above comment. Leave it as "utf-8" if possible. If not, please ping me on Teams and explain why it's not possible.

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "write_output": true,
     "output_folder_path": "./output/APAC_Taiwan/",
     "output_file_name_prefix": "Combined_Taiwan_Investment_20200501_20200531_",
+    "output_encoding" : "utf-8-sig",
```
Comment (plartoo) [inline data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json:14]: Leave it as "utf-8"?

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json (hunk 2 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json:15]: Don Phyo, I tested it leaving only utf-8 and it didn't work, and then I tried to leave the utf-8-sing and it worked correctly

File: data_architect_misc/data_transformer/configs/APAC_Vietnam/step1_transform_raw_sos_investment_data.json [hunk 5]
```diff
@@ -48,12 +48,12 @@
       {
         "__function_comment__": "Delete row with value 'Total' from Date column.",
         "function_name": "drop_rows_with_matching_string_values",
-        "function_args": [["DATE", "SECTOR", "BRAND"],[["Total"], ["Total"], ["Total"]]]
+        "function_args": [["DATE", "SECTOR", "BRAND"],["Total","Total","Total"]]
```
Comment (plartoo) [inline data_architect_misc/data_transformer/configs/APAC_Vietnam/step1_transform_raw_sos_investment_data.json:51]: Why did we removed square brackets in ["Total"]? This is designed so that if we want to delete rows with, for example, 'Total' or 'Sum' in 'Brand' column, we can do something like this:
"function_args": [["DATE", "SECTOR", "BRAND"], [["Total"], ["Total"], ["Total", "Sum"]] ]

Please revert the change to the function if you updated its method signature.

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json (hunk 3 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json:15]: if I left the utf-8 configured in the config file, the output file showed up with strange character but with the utf-8-sig the characters appears correctly

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json (hunk 4 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json:14]: It's no possible sire or we should find another soluction

File: data_architect_misc/data_transformer/configs/APAC_Vietnam/step1_transform_raw_sos_investment_data.json (hunk 5 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/configs/APAC_Vietnam/step1_transform_raw_sos_investment_data.json:51]: This is because the function that we use to delete all these strings, it's not using a list of values because of function uses a str.contain and this funtion no requires a list as parameter

File: data_architect_misc/data_transformer/transform_functions/apac_singapore_transform_functions.py [hunk 6]
```diff
@@ -36,7 +36,7 @@ def __init__(self, config):
         # specific category mapping for this country
         self.category_mappings = comp_harm_constants.ENGLISH_CATEGORY_MAPPINGS
 
-    def add_RAW_BRAND_column_for_multiple_files_using_specific_string_part_of_filename(
+    def create_new_dataframe_and_add_RAW_BRAND_column_for_multiple_files_using_specific_string_part_of_filename(
```
Comment (plartoo) [inline data_architect_misc/data_transformer/transform_functions/apac_singapore_transform_functions.py:39]: Please rename this method to: "create_new_dataframe_with_RAW_BRAND_column_from_multiple_input_file_with_brand_names_as_part_of_the_file_names"

File: data_architect_misc/data_transformer/configs/AED_Ukraine/config.json (hunk 1 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/configs/AED_Ukraine/config.json:9]: Yes sire, you are right, I will delete this line of the config file

File: data_architect_misc/data_transformer/transform_functions/apac_vietnam_transform_functions.py [hunk 7]
```diff
@@ -39,11 +39,11 @@ def create_new_dataframe_from_given_sheet_names_and_add_media_type_column_using_
 
             # Create  RAW_MEDIA_TYPE column based on sheet names.
             temp_df[RAW_MEDIA_TYPE_COLUMN] = sheet
+            temp_df.dropna(subset=["Date", "Sector", "Brand"],inplace=True)
```
Comment (plartoo) [inline data_architect_misc/data_transformer/transform_functions/apac_vietnam_transform_functions.py:42]: Please rename this method (don't forget to update the config file!) like:
"create_new_dataframe_with_RAW_MEDIA_TYPE_column_from_input_file_with_multiple_sheets_dedicated_to_each_media_type"

File: data_architect_misc/data_transformer/transform_functions/apac_singapore_transform_functions.py (hunk 6 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/transform_functions/apac_singapore_transform_functions.py:39]: change done

File: data_architect_misc/data_transformer/transform_functions/apac_vietnam_transform_functions.py (hunk 7 above)
Comment (Maik8080) [reply data_architect_misc/data_transformer/transform_functions/apac_vietnam_transform_functions.py:42]: Done

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json (hunk 4 above)
Comment (plartoo) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step3_combine_tranformed_noncompete_and_compete_data.json:14]: Please leave a comment before "output_encoding" like below:
"__output_encoding_comment__": "In Taiwan input files, we have to use utf-8-sig as output encoding or else it does not work. The character outputs become incorrect if we use utf-8."

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json (hunk 3 above)
Comment (plartoo) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step2_transform_sos_raw_investment_data.json:15]: Please leave a comment before "output_encoding" like below:
"__output_encoding_comment__": "In Taiwan input files, we have to use utf-8-sig as output encoding or else it does not work. The character outputs become incorrect if we use utf-8."

File: data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json (hunk 2 above)
Comment (plartoo) [reply data_architect_misc/data_transformer/configs/APAC_Taiwan/step1_transform_raw_investment_data.json:15]: Please leave a comment before "output_encoding" like below:
"__output_encoding_comment__": "In Taiwan input files, we have to use utf-8-sig as output encoding or else it does not work. The character outputs become incorrect if we use utf-8."

**Review by plartoo — APPROVED** [review]
